### PR TITLE
Task J-16: add checkbox markup

### PR DESCRIPTION
**Pull request by artem-totality**

### CODE REVIEW

File: frontend/src/common/enums/ui/checkbox-style.enum.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+enum CheckboxStyle {
+  CHECKBOX_SIGN_IN = 'checkboxSignIn',
+}
+
+export { CheckboxStyle };
```
Comment (ghost) [inline frontend/src/common/enums/ui/checkbox-style.enum.ts]:
```suggestion
```
we have only one checkbox type

File: frontend/src/common/enums/enums.ts [hunk 2]
```diff
@@ -2,5 +2,5 @@ export { ApiPath, UsersApiPath, AuthApiPath } from './api/api';
 export { AppRoute, ENV, DataStatus } from './app/app';
 export { HttpHeader, HttpMethod } from './http/http';
 export { ContentType } from './file/file';
-export { ButtonType, InputType } from './ui/ui';
+export { CheckboxStyle, ButtonType, InputType } from './ui/ui';
```
Comment (ghost) [inline frontend/src/common/enums/enums.ts]:
```suggestion
export { ButtonType, InputType } from './ui/ui';
```

File: frontend/src/components/common/checkbox/checkbox.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import { CheckboxStyle, InputType } from 'common/enums/enums';
+import styles from './styles.module.scss';
+
+type Props = {
+  label: string;
+  onChange: (evt: React.ChangeEvent<HTMLInputElement>) => void;
+  name: string;
+  useStyle?: CheckboxStyle;
```
Comment (ghost) [inline frontend/src/components/common/checkbox/checkbox.tsx]:
```suggestion
```

File: frontend/src/common/enums/ui/ui.ts [hunk 4]
```diff
@@ -1,2 +1,3 @@
+export { CheckboxStyle } from './checkbox-style.enum';
```
Comment (ghost) [inline frontend/src/common/enums/ui/ui.ts]:
```suggestion
```

File: frontend/src/components/common/checkbox/checkbox.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+const Checkbox: React.FC<Props> = ({
+  useStyle = '',
+  isRequire = false,
+  isDisabled = false,
+  isChecked = false,
+  label,
+  name,
+  onChange,
+}) => (
+  <label className={`${styles.checkbox} ${styles[useStyle]}`}>
```
Comment (ghost) [inline frontend/src/components/common/checkbox/checkbox.tsx]:
```suggestion
  <label className={styles.checkbox}>
```

File: frontend/src/components/common/checkbox/styles.module.scss [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  span {
+    position: relative;
+    display: inline-block;
+    padding-left: 27px;
+    font-family: 'Raleway', sans-serif;
+    font-size: 14px;
+    font-weight: 500;
+    line-height: 1.35;
+    user-select: none;
+    color: var(--color-primary-dark);
```
Comment (ghost) [inline frontend/src/components/common/checkbox/styles.module.scss]: do all this things directly on an input

File: frontend/src/components/common/checkbox/styles.module.scss [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+      width: 15px;
+      height: 15px;
+      left: 0;
+      top: 1px;
+      border: 1px solid var(--color-button-background);
+      border-radius: 2px;
+    }
+  }
+  input {
+    display: none;
```
Comment (ghost) [inline frontend/src/components/common/checkbox/styles.module.scss]:
```suggestion
appearance: none;
```

File: frontend/src/components/common/checkbox/styles.module.scss [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    color: #717477;
+  }
+}
+
+.checkbox {
+  span {
+    position: relative;
+    display: inline-block;
+    padding-left: 27px;
+    font-family: 'Raleway', sans-serif;
```
Comment (ghost) [inline frontend/src/components/common/checkbox/styles.module.scss]: add it to global styles on `body`

**Review by ghost — CHANGES_REQUESTED** [review]

File: frontend/src/components/common/checkbox/checkbox.tsx [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+import styles from './styles.module.scss';
+
+type Props = {
+  label: string;
+  onChange: (evt: React.ChangeEvent<HTMLInputElement>) => void;
+  name: string;
+  useStyle?: CheckboxStyle;
+  isRequire?: boolean;
+  isDisabled?: boolean;
+  isChecked?: boolean;
```
Comment (ghost) [inline frontend/src/components/common/checkbox/checkbox.tsx]:
```suggestion
  isChecked: boolean;
```

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — DISMISSED** [review]
LGTM

**Review by ghost — DISMISSED** [review]

File: frontend/src/components/common/checkbox/checked.svg [hunk 10]
```diff
@@ -0,0 +1,3 @@
+<svg width="9" height="11" viewBox="0 0 9 11" fill="none" xmlns="http://www.w3.org/2000/svg">
+<path d="M4.29382 8.20027L4.29372 8.20037C4.20775 8.28685 4.09032 8.33521 3.96821 8.33521C3.84609 8.33521 3.72866 8.28685 3.64269 8.20037L3.64259 8.20027L0.759741 5.31696L0.759738 5.31695C0.480069 5.03728 0.480117 4.58383 0.759705 4.30477L0.759742 4.30473L1.12071 3.94367C1.12071 3.94367 1.12071 3.94367 1.12072 3.94366C1.40047 3.66401 1.85338 3.66402 2.13302 3.94367L3.93285 5.74359L3.9682 5.77895L4.00356 5.74359L8.86695 0.880105C9.14672 0.600429 9.60008 0.600476 9.87924 0.880072L9.87926 0.880095L10.2402 1.24116L10.2756 1.20581L10.2402 1.24117C10.5199 1.52083 10.5199 1.97419 10.2403 2.25336L10.2402 2.25338L4.29382 8.20027Z" fill="white" stroke="#F3427F" stroke-width="0.1"/>
+</svg>
```
Comment (liza-veis) [inline frontend/src/components/common/checkbox/checked.svg]: Maybe it would be better to put it in the assets/img

Comment (ghost) [reply frontend/src/components/common/checkbox/checked.svg]: Good catch!

Agreed

**Review by yevhenprybytok — DISMISSED** [review]

File: frontend/src/components/common/checkbox/styles.module.scss [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+      width: 15px;
+      height: 15px;
+      padding-top: 1px;
+      text-align: center;
+      border: 1px solid var(--color-button-background);
+      border-radius: 2px;
+    }
+    &:checked {
+      &::before {
+        content: url('../../../assets/img/checked.svg');
```
Comment (ghost) [inline frontend/src/components/common/checkbox/styles.module.scss:22]:
```suggestion
        content: url('assets/img/checked.svg');
```
should also like this

**Review by ghost — CHANGES_REQUESTED** [review]

File: frontend/src/components/common/checkbox/styles.module.scss [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+  color: var(--color-primary-dark);
+  input {
+    position: relative;
+    appearance: none;
+    &::before {
+      content: '';
+      position: absolute;
+      right: 7px;
+      top: -9px;
+      box-sizing: border-box;
```
Comment (ghost) [inline frontend/src/components/common/checkbox/styles.module.scss]:
```suggestion
```

File: frontend/src/components/common/checkbox/styles.module.scss [hunk 13]
```diff
@@ -0,0 +1,33 @@
+.checkbox {
+  display: inline-block;
+  padding-left: 27px;
+  font-size: 14px;
+  font-weight: 500;
+  line-height: 1.35;
```
Comment (ghost) [inline frontend/src/components/common/checkbox/styles.module.scss]:
```suggestion
```

File: frontend/src/components/common/checkbox/styles.module.scss [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+    &::before {
+      content: '';
+      position: absolute;
+      right: 7px;
+      top: -9px;
+      box-sizing: border-box;
+      width: 15px;
+      height: 15px;
+      padding-top: 1px;
+      text-align: center;
```
Comment (ghost) [inline frontend/src/components/common/checkbox/styles.module.scss]:
```suggestion
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: frontend/src/components/common/checkbox/styles.module.scss (hunk 11 above)
Comment (artem-totality) [reply frontend/src/components/common/checkbox/styles.module.scss:22]: it doesn't find such route

Comment (ghost) [reply frontend/src/components/common/checkbox/styles.module.scss:22]: okay

**Review by ghost — APPROVED** [review]

**Pull request merged**